numpy/random/mt19937.pyx

```diff
@@ -59,6 +61,14 @@ cdef class MT19937:
         ``/dev/urandom`` (or the Windows analog) if available. If unavailable,
         a 32-bit hash of the time and process ID is used.
 
+    Attributes
+    ----------
+    lock: threading.Lock
+        Lock instance that is shared so that the same bit git generator can
+        be used in multiple Generators without corrupting the state. Code that
+        generates values from a bit generator should hold the bit generator's
+        lock.
+
```

Comment:
It would be nice to refactor the BitGenerators to inherit from a common base class so all this does not need repeating

Reply:
I agree.  I think the challenge here is the limitation imposed by Cython extension types since you can't do all of the Python doc string fixes after the fact.  n times copy-pasta is a bit soul destroying

Would probably need to use composition instead of inheritance where a small extension class provides the core functionality to a larger Python class, which could then avoid repeats.

Reply:
never mind, this already appears as a follow-on task in #13164